internal/backends/testing/search_api.rs

```diff
@@ -705,3 +845,92 @@ fn test_conditional() {
     assert_eq!(ElementHandle::find_by_element_id(&app, "App::visible-element").count(), 1);
     assert_eq!(ElementHandle::find_by_element_id(&app, "App::inner-element").count(), 1);
 }
+
+#[test]
+fn test_matches() {
+    crate::init_no_event_loop();
+
+    slint::slint! {
+        component Base inherits Rectangle {}
+
+        export component App inherits Window {
+            in property <bool> condition: false;
+            if condition: dynamic-elem := Base {
+                accessible-role: text;
+            }
+            visible-element := Rectangle {
+                visible: !condition;
+                inner-element := Text { text: "hello"; }
+            }
+        }
+    }
+
+    let app = App::new().unwrap();
+
+    let root = app.root_element();
+
+    assert_eq!(root.match_descendants().match_inherits("Rectangle").find_all().len(), 1);
+    assert_eq!(root.match_descendants().match_inherits("Base").find_all().len(), 0);
+    assert!(root.match_descendants().match_id("App::dynamic-elem").find_first().is_none());
+
+    assert_eq!(root.match_descendants().match_id("App::visible-element").find_all().len(), 1);
+    assert_eq!(root.match_descendants().match_id("App::inner-element").find_all().len(), 1);
+
+    assert_eq!(
+        root.match_descendants()
+            .match_id("App::visible-element")
+            .match_descendants()
+            .match_accessible_role(crate::AccessibleRole::Text)
+            .find_first()
+            .and_then(|elem| elem.accessible_label())
+            .unwrap_or_default(),
+        "hello"
+    );
+
+    app.set_condition(true);
+
+    assert!(root
+        .match_descendants()
+        .match_id("App::visible-element")
+        .match_descendants()
+        .match_accessible_role(crate::AccessibleRole::Text)
+        .find_first()
+        .is_none());
+
+    let elems = root.match_descendants().match_id("App::dynamic-elem").find_all();
+    assert_eq!(elems.len(), 1);
+    let elem = &elems[0];
+
+    assert_eq!(elem.id().unwrap(), "App::dynamic-elem");
+    assert_eq!(elem.type_name().unwrap(), "Base");
+    assert_eq!(elem.bases().unwrap().count(), 1);
+    assert_eq!(elem.accessible_role().unwrap(), crate::AccessibleRole::Text);
+
+    assert_eq!(root.match_descendants().match_inherits("Base").find_all().len(), 1);
+}
+
+/*
```

Comment:
why is this commented out?

Reply:
This is a test that assumes the availability of children traversal, which I removed again. I can remove the commented out test, but I want to bring it back when adding children support.